main.go

```diff
@@ -58,16 +62,76 @@ func handleJump(args []string, provider string) []string {
 	return ssh_args_str
 }
 
-func selectHost(hosts []cr.Host) cr.Host {
+func hostMatch(f *cr.Host, filterstr []string) bool {
```

Comment:
This function is too complex

Reply:
Added more comments but not sure can be made clearer any other way ;-)